apps/argos_aggregation/lib/argos_aggregation/application.ex

```diff
@@ -31,45 +32,54 @@ defmodule ArgosAggregation.Application do
   end
 
   def put_index() do
-    "#{@elasticsearch_url}"
-    |> HTTPoison.put()
+    Finch.build(:put, "#{@elasticsearch_url}")
+    |> Finch.request(ArgosFinch)
   end
 
   def delete_index() do
-    "#{@elasticsearch_url}"
-    |> HTTPoison.delete()
+    Finch.build(:delete, "#{@elasticsearch_url}")
+    |> Finch.request(ArgosFinch)
   end
 
 
   def put_mapping() do
     mapping = File.read!(@elasticsearch_mapping_path)
-    "#{@elasticsearch_url}/_mapping"
-    |> HTTPoison.put(mapping, [{"Content-Type", "application/json"}])
+    Finch.build(:put, "#{@elasticsearch_url}/_mapping", [{"Content-Type", "application/json"}],mapping)
+    |> Finch.request(ArgosFinch)
   end
 
 
   defp initialize_index() do
-    case HTTPoison.get(@elasticsearch_url) do
+    res =Finch.build(:put, @elasticsearch_url)
+    |> Finch.request(ArgosFinch)
+    case res do
```

Comment:
Das seeding Problem entsteht in diesem Case:
Der Endpoint Antwortet mit 
`{:ok, %Finch.Response{body: "{\"error\":{\"root_cause\":[{\"type\":\"resource_already_exists_exception\"...`
Dafür gibt es aber keinen Case Fall und damit steigt Argos mit dem Fehler 
`(CaseClauseError) no case clause matching:`
aus.

Das case braucht glaube ich insgesamt noch etwas liebe, es ist nur sehr schwer da durchzusteigen.

Reply:
aber warum hat es mit httpoison davor funktionniert ?
da gab es auch schon kein case für "already exists"